src/mame/drivers/megadriv_vt_hybrid.cpp

```diff
@@ -0,0 +1,163 @@
+		// Mega Drive part
+		: md_base_state(mconfig, type, tag),
+		m_md_is_running(false),
+		m_bank(0),
+		m_rom(*this, "maincpu")
+	{}
+
+	// Mega Drive part
+	DECLARE_MACHINE_START(megadriv_vt0203);
+	uint16_t read(offs_t offset);
+	void megadriv_vt0203_pal(machine_config &config);
+	void megadriv_vt0203_map(address_map &map);
+
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+private:
+	bool m_md_is_running;
+
+	// Mega Drive part
+	int m_bank;
+	required_region_ptr<uint16_t> m_rom;
+	uint32_t screen_update_hybrid(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+	DECLARE_WRITE_LINE_MEMBER(screen_vblank_hybrid);
+
+};
+
+
+// todo, use actual MD map, easier once maps are part of base class.
+void megadriv_vt0203_state::megadriv_vt0203_map(address_map &map)
+{
+	map(0x000000, 0x3fffff).r(FUNC(megadriv_vt0203_state::read)); /* Cartridge Program Rom */
+	map(0xa00000, 0xa01fff).rw(FUNC(megadriv_vt0203_state::megadriv_68k_read_z80_ram), FUNC(megadriv_vt0203_state::megadriv_68k_write_z80_ram));
+	map(0xa02000, 0xa03fff).w(FUNC(megadriv_vt0203_state::megadriv_68k_write_z80_ram));
+	map(0xa04000, 0xa04003).rw(FUNC(megadriv_vt0203_state::megadriv_68k_YM2612_read), FUNC(megadriv_vt0203_state::megadriv_68k_YM2612_write));
+	map(0xa06000, 0xa06001).w(FUNC(megadriv_vt0203_state::megadriv_68k_z80_bank_write));
+	map(0xa10000, 0xa1001f).rw(FUNC(megadriv_vt0203_state::megadriv_68k_io_read), FUNC(megadriv_vt0203_state::megadriv_68k_io_write));
+	map(0xa11100, 0xa11101).rw(FUNC(megadriv_vt0203_state::megadriv_68k_check_z80_bus), FUNC(megadriv_vt0203_state::megadriv_68k_req_z80_bus));
+	map(0xa11200, 0xa11201).w(FUNC(megadriv_vt0203_state::megadriv_68k_req_z80_reset));
+	map(0xc00000, 0xc0001f).rw(m_vdp, FUNC(sega315_5313_device::vdp_r), FUNC(sega315_5313_device::vdp_w));
+	map(0xe00000, 0xe0ffff).ram().mirror(0x1f0000).share("megadrive_ram");
+}
+
+uint16_t megadriv_vt0203_state::read(offs_t offset)
+{
+	return m_rom[(((m_bank * 0x10000) + (offset << 1)) & (0x400000 - 1))/2];
+}
+
+
+// controller is wired directly into unit, no controller slots
+static INPUT_PORTS_START( megadriv_vt0203 )
+	PORT_INCLUDE( md_common )
+
+	PORT_MODIFY("PAD1")
+#if 0
+	PORT_BIT( 0x0010, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(1)
+	PORT_BIT( 0x0020, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(1)
+	PORT_BIT( 0x0040, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(1)
+	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_START1 )
+#else
+	PORT_BIT( 0x00ff, IP_ACTIVE_LOW, IPT_UNUSED )
+#endif
+
+	PORT_MODIFY("PAD2")
+#if 0
+	PORT_BIT( 0x0010, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(2)
+	PORT_BIT( 0x0020, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(2)
+	PORT_BIT( 0x0040, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(2)
+	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_START2 )
+##else
+	PORT_BIT( 0x00ff, IP_ACTIVE_LOW, IPT_UNUSED )
+#endif
+INPUT_PORTS_END
+
+
+void megadriv_vt0203_state::machine_start()
+{
+	printf("machine_start\n");
```

Comment:
Remember to change this to `logerror` or one of the things from logmacro.h.

Reply:
ah yeah, missed that, will fix